src/fpm_targets.f90

```diff
@@ -238,6 +240,17 @@ subroutine build_target_list(targets,model)
                         call add_dependency(targets(1)%ptr, targets(size(targets))%ptr)
                     end if
 
+                case (FPM_UNIT_CPPSOURCE) 
+
+                    call add_target(targets,package=model%packages(j)%name,source = sources(i), &
+                                type = FPM_UNIT_CPPSOURCE, &
+                                output_name = get_object_name(sources(i)), &
+                                macros = model%packages(j)%macros, &
+                                version = model%packages(j)%version)
+
+                    !> Add stdc++ as a linker flag.
+                    model%link_flags = model%link_flags // "stdc++"
```

Comment:
This needs to be added to `model%link_librarys` and we only want to add it once (not repeatedly for each cpp file).

```suggestion
                    if (.not. ("stdc++" .in. model%link_libraries)) then
                        model%link_libraries = [model%link_libraries, string_t("stdc++")]
                    end if
```